### PR TITLE
1695 grad cam revival

### PR DESCRIPTION
**Pull request by jasminziegler**
# Description

[Component Requirement #1695](https://dev.azure.com/cgmorg/ChildGrowthMonitor/_workitems/edit/1695) 

# Motivation:

GradCAM lets us display heatmaps to find out on which parts of the scans our models focus on during training. This can serve as a sanity check for our models.

# What has been done?

This PR delivers a notebook in cgm-ml which downloads a pretrained model and a dataset. From this dataset, some examples have been chosen to compute a heatmap and overlay it on the original image. 

1. First try: revive existing gradcam code, stopped this after several days due to tf/keras version issues that could not been resolved.
2. Second try: adapt [F. Chollet example ](https://colab.research.google.com/github/keras-team/keras-io/blob/master/examples/vision/ipynb/grad_cam.ipynb#scrollTo=GMxN5Ec63x9E) to our use case.
- change model to our pretrained model - download from azureml
- change data from example jpeg file to depthmap example from anon-depthmap-mini (get familiar with depthmap toolkit)

# Next steps
[Feature 36](https://dev.azure.com/cgmorg/ChildGrowthMonitor/_workitems/edit/36) aims for an integration of GradCAM in cgm-rg by adding a workflow that outputs a heatmap image to the tagging tool.

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] My Code is review by 1 People

### CODE REVIEW

**Comment by jasminziegler** [comment]
Hey @prajurock, could you pls quickly review?
This is "only" a POC actually, so only a test notebook :-) nothing has been changed elsewhere.
This will be continued in a new CR/PR tackling the integration of this POC in cgm-rg.

Thanks amigo!

File: .github/workflows/continous-integration.yml [hunk 1]
```diff
@@ -16,16 +16,9 @@ jobs:
...
         python-version: ${{ matrix.python-version }}
-    - uses: actions/cache@v2
-      id: cache-venv
-      with:
-        path: ./.venv/
-        key: ${{ runner.os }}-cache-v3-${{ hashFiles('requirements.txt') }}
-        restore-keys: |
-          ${{ runner.os }}-cache-v3-
-    - run: python -m venv ./.venv && . ./.venv/bin/activate &&
+    - name: test
```
Comment (prajurock) [inline .github/workflows/continous-integration.yml:19]: undo this change and git merge main

File: requirements.txt [hunk 2]
```diff
@@ -26,4 +26,4 @@ pep8-naming==0.12.1
 pytest==6.1.2
 pytest-cov==2.10.1
 pytest-azurepipelines==0.8.0
-yacs
+yacs==0.1.8
```
Comment (prajurock) [inline requirements.txt:29]: remove this too.

**Review by prajurock — APPROVED** [review]
If it has a script and you only do use visualisation in a notebook that would be better.
But as you say its POC I am approving it and hoping it will get better.

File: .github/workflows/continous-integration.yml (hunk 1 above)
Comment (jasminziegler) [reply .github/workflows/continous-integration.yml:19]: @prajurock Thanks for the quick review and the hint, good point.
I will re-write it in the follow-up so the code will be in a script and only visualisation in notebook.

**Pull request merged**